### PR TITLE
Replace listdir with scandir

### PR DESCRIPTION
**Pull request by hypothesist**
`scandir` is faster than listdir with pretty much the same API

Closes https://github.com/coala-analyzer/coala/issues/2319

### CODE REVIEW

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2322/commits/a2f497096ca57af6c5946dfde73a57254a4d1765

**Comment by sils1297** [comment]
unack a2f4970 , let's put the import in Future

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2322/commits/2b01666c9db40a16636077872097ee5532aa2c49

File: coalib/misc/Future.py [hunk 1]
```diff
@@ -79,5 +79,11 @@ def __get__(self, obj, cls):
         @property
         def __isabstractmethod__(self):
             return getattr(self.func, "__isabstractmethod__", False)
+elif sys.version_info < (3, 5):
+    from functools import partialmethod
+    from scandir import scandir
 else:
+    # Start ignoring PyUnusedCodeBear
+    from os import scandir
```
Comment (sils1297) [inline coalib/misc/Future.py]: you don't import scandir if version < 3.4

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2322/commits/b59d987a2002fe0cf7beaee85090b37ca4471bba#diff-ba542729717668e8dd1a9cde8a92209dR85

**Comment by hypothesist** [comment]
@rultor merge

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2322/commits/da6c3ebdf7034a13dc31400750006a815c4c2de6

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**